src/context-parser-handlebars.js

```diff
@@ -404,6 +430,49 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
 
                 output += t.output;
 
+            // if the 'partial expression' is not in Data State,
+            // TODO: we support basic partial only, need to enhance it.
+            // http://handlebarsjs.com/partials.html
+            } else if (node.type === handlebarsUtils.PARTIAL_EXPRESSION && 
+                parser.getCurrentState() !== stateMachine.State.STATE_DATA && 
+                (re = handlebarsUtils.isValidExpression(node.content, 0, node.type)) && 
+                (partialName = re.tag)) {
+
+                key = this._config._namespace === ''? partialName : this._config._namespace + '/' + partialName;
+                partialContent = this._config._partialsCache[key];
+                if (partialContent === undefined) {
+                    msg = "[ERROR] SecureHandlebars: Fail to read the " + node.content + " partial template! ";
+                    exceptionObj = new ContextParserHandlebarsException(msg, this._fileName, this._lineNo, this._charNo);
+                    handlebarsUtils.handleError(exceptionObj, true);
```

Comment:
we have to discuss whether we want to throw an error here.
people might have used `registerPartial()` and leave `partialsCache` empty. but then an error is thrown whenever a partial expression is encountered. 

I propose to fallback to the original warning (in the so-called non-HTML state)